src/main/scala/is/hail/sparkextras/OrderedRDD2.scala

```diff
@@ -14,15 +14,15 @@ class OrderedDependency(left: OrderedRVD, right: OrderedRVD) extends NarrowDepen
 
 object OrderedDependency {
   def getDependencies(p1: OrderedRVDPartitioner, p2: OrderedRVDPartitioner)(partitionId: Int): Range = {
-    val lastPartition = if (partitionId == p1.rangeBounds.length)
+    val lastPartition = if (partitionId == p1.rangeBounds.length - 1)
```

Comment:
This can be cleaned up, too.  Instead of checking ends, etc., you can just return rangeTree.queryValues(p1.rangeBounds(partitionId)).

Reply:
You can't do this either: no IntervalTree search on intervals.  I'd make this match the code in filterIntervals (empty is also an option here).

Reply:
yeah, oof. the way this was written was pretty silly and potentially included a lot of unnecessary stuff.

Reply:
the way I fixed it up, I mean